extension/chrome/elements/attachment.ts

```diff
@@ -320,10 +327,13 @@ export class AttachmentDownloadView extends View {
     });
     Xss.sanitizeRender(this.downloadButton, this.originalButtonHTML || '');
     if (result.success) {
+      const attachmentForSave = new Attachment({ name: result.filename, type: this.attachment.type, data: result.content });
       if (!result.filename || ['msg.txt', 'null'].includes(result.filename)) {
         result.filename = this.attachment.name;
       }
-      Browser.saveToDownloads(new Attachment({ name: result.filename, type: this.attachment.type, data: result.content }));
+      if (await AttachmentWarnings.confirmSaveToDownloadsIfNeeded(attachmentForSave, this)) {
```

Comment:
Should we add this check to more places, e.g. `ManualDecryptView.decryptAndDownload`?

Reply:
Yes, I think we should add this check for all places where user can download attachment.

What about adding parameter `shouldCheckAttachmentType` to `Browser.saveToDownloads`?

Reply:
I was thinking about it too, but this would make `saveToDownloads` `async` / return a promise. This may be not a problem, in some cases we can run it with `Catch.try`, but I'm a bit bothered by this piece of code
https://github.com/FlowCrypt/flowcrypt-browser/blob/14ffde42a5bc8ca13981c2138bff58c7f1aa7251/extension/js/common/settings.ts#L466

This
```
await Browser.saveToDownloads(new Attachment({ name, type: 'text/plain', data: Buf.fromUtfStr(backupText) }));
await Ui.delay(1000);
```
would look ugly (as the first `await` only awaits for the possible confirmation to complete, and the second `await` await the handling of the message -- that's counter-intuitive) 
Perhaps, we could include `await Ui.delay(1000);` into `Browser.saveToDownloads`, @sosnovsky ?

Reply:
Having `Ui.delay` in `Browser.saveToDownloads` will affect all downloads, while it's needed only in this single case, so it's not the best solution.

What about moving this delay to `resetAccount` right after `await Settings.collectInfoAndDownloadBackupFile(acctEmail)`?
There we can also add comment `// wait before backup download is finished`

Reply:
My question was: should we make `Browser.saveToDownloads` `async` or not?
My point is: if it's not async, then it's clear that the function only initiates some process and we should (in some cases) give it some time to finish.
If it's async, and we await it, it's not evident that the promise is resolved "half-way through", and we need to wait "a bit more" still. This design is a bit obscure.

> Having Ui.delay in Browser.saveToDownloads will affect all downloads,

As I see it, with `Catch.try(() => Browser.saveToDownloads)` the resources will be released one second later, that's all.
Most of the calls are initiated by event handlers, so `Catch.try` isn't required there.